proxy/src/main/java/com/velocitypowered/proxy/connection/backend/BackendPlaySessionHandler.java

```diff
@@ -68,12 +69,26 @@ public void handle(MinecraftPacket packet) {
                 return;
             }
 
+            if (!connection.hasCompletedJoin() && pm.getChannel().equals(VelocityConstants.FORGE_LEGACY_HANDSHAKE_CHANNEL)) {
+                if (!connection.isModded()) {
+                    connection.setModded(true);
+
+                    // We must always reset the handshake before a modded connection is established if
+                    // we haven't done so already.
+                    connection.getPlayer().sendLegacyForgeHandshakeResetPacket();
+                }
+
+                // Always forward these messages during login
+                connection.getPlayer().getConnection().write(pm);
+                return;
+            }
+
             MessageHandler.ForwardStatus status = server.getChannelRegistrar().handlePluginMessage(connection,
                     ChannelSide.FROM_SERVER, pm);
             if (status == MessageHandler.ForwardStatus.FORWARD) {
                 connection.getPlayer().getConnection().write(pm);
             }
-        } else {
+        } else if (connection.hasCompletedJoin()) {
```

Comment:
You need to add this particular check to `handleUnknown()` too. I believe this is why you're getting this error but I haven't checked.

Reply:
Yep, that would make sense - if you don't get to testing this tonight, I'll try to do that tomorrow.

Reply:
Actually, thinking about it, this particular packet should be discarded during join. I think it's a packet from the client referring to the position on the old server it is on (that has since been disconnected), we haven't told the client to switch worlds yet.

So in this case, we probably need to discard, not queue. During the handshake, no other packets should be coming from the client anyway.